src/Netatmo/CredentialManager.cs

```diff
@@ -44,6 +46,18 @@ public async Task GenerateToken(string username, string password, Scope[] scopes
             CredentialToken = new CredentialToken(token, clock);
         }
 
+        public async Task GenerateToken(string oauth2Token)
```

Comment:
You don't generate, token but you provide it
```suggestion
        public async Task ProvideOAuth2Token(string oauth2Token)
```